### PR TITLE
Endpoint refactoring

### PR DESCRIPTION
**Pull request by Monomus**

### CODE REVIEW

File: src/main/java/io/swagslash/gametrackerserver/dto/webapp/GameDTO.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
+package io.swagslash.gametrackerserver.dto.webapp;
+
+/**
+ * DTO for delivering data to Frontend
+ */
+public class GameDTO {
```
Comment (W3D3) [inline src/main/java/io/swagslash/gametrackerserver/dto/webapp/GameDTO.java:10]: This should include gamemodes and genres?

File: src/main/java/io/swagslash/gametrackerserver/service/impl/GameServiceImpl.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        for (Game game : games) {
+            dtos.add(entityToDTO(game));
+        }
+
+        return dtos;
+    }
+
+    @Override
+    public List<GameDTO> findBySearch(String term) {
+        IGDBWrapper igdbWrapper = new IGDBWrapper(appProperties.getIgdb().getKey());
```
Comment (W3D3) [inline src/main/java/io/swagslash/gametrackerserver/service/impl/GameServiceImpl.java]: wrap calls to this wrapper in try catch
also maybe use the IGDBApi interface to make it more general

File: src/main/java/io/swagslash/gametrackerserver/config/AppProperties.java [hunk 3]
```diff
@@ -5,12 +5,18 @@
...
 
+/**
+ * Class for accessing application propertiers speficied in Application.yaml
+ */
 @ConfigurationProperties(prefix = "app")
 public class AppProperties {
+
     private final Auth auth = new Auth();
     private final OAuth2 oauth2 = new OAuth2();
+    private final IGDB igdb = new IGDB();
```
Comment (W3D3) [inline src/main/java/io/swagslash/gametrackerserver/config/AppProperties.java:16]: very confusing naming, what is a "igdb"?

File: src/main/java/io/swagslash/gametrackerserver/config/AppProperties.java [hunk 4]
```diff
@@ -44,11 +50,27 @@ public OAuth2 authorizedRedirectUris(List<String> authorizedRedirectUris) {
...
     }
 
     public OAuth2 getOauth2() {
         return oauth2;
     }
+
+    public IGDB getIgdb() {
+        return igdb;
+    }
 }
```
Comment (GustavAT) [inline src/main/java/io/swagslash/gametrackerserver/config/AppProperties.java]: Please add a new-line

File: src/main/java/io/swagslash/gametrackerserver/config/SecurityConfig.java [hunk 5]
```diff
@@ -22,6 +22,10 @@
 import org.springframework.security.crypto.password.PasswordEncoder;
 import org.springframework.security.web.authentication.UsernamePasswordAuthenticationFilter;
 
+/**
+ * Security Configuration for Spring Boot
+ * If api endpoint should be available wihtout Authentication specify Exception in this file
```
Comment (GustavAT) [inline src/main/java/io/swagslash/gametrackerserver/config/SecurityConfig.java]: typo `without`

File: src/main/java/io/swagslash/gametrackerserver/config/SecurityConfig.java [hunk 6]
```diff
@@ -83,50 +85,50 @@ public AuthenticationManager authenticationManagerBean() throws Exception {
...
+                    "/favicon.ico",
+                    "/**/*.png",
+                    "/**/*.gif",
+                    "/**/*.svg",
+                    "/**/*.jpg",
+                    "/**/*.html",
+                    "/**/*.css",
+                    "/**/*.js")
+            .permitAll()
+            .antMatchers("/api/auth/**", "/api/oauth2/**", "/api/agent/**")
```
Comment (GustavAT) [inline src/main/java/io/swagslash/gametrackerserver/config/SecurityConfig.java]: I think you should leave this to just `/oauth2/**` as you don't specify `api` in your oauth base uri below

File: src/main/java/io/swagslash/gametrackerserver/config/SecurityConfig.java [hunk 7]
```diff
@@ -83,50 +85,50 @@ public AuthenticationManager authenticationManagerBean() throws Exception {
...
+            .permitAll()
+            .anyRequest()
+            .authenticated()
+            .and()
+            .oauth2Login()
+            .authorizationEndpoint()
+            .baseUri("/oauth2/authorize")
```
Comment (GustavAT) [inline src/main/java/io/swagslash/gametrackerserver/config/SecurityConfig.java:121]: Might not work with the matcher specified above

File: src/main/java/io/swagslash/gametrackerserver/config/SecurityConfig.java [hunk 8]
```diff
@@ -83,50 +85,50 @@ public AuthenticationManager authenticationManagerBean() throws Exception {
...
+            .authorizationRequestRepository(cookieAuthorizationRequestRepository())
+            .and()
+            .redirectionEndpoint()
+            .baseUri("/oauth2/callback/*")
```
Comment (GustavAT) [inline src/main/java/io/swagslash/gametrackerserver/config/SecurityConfig.java:125]: Might not work with the matcher specified above

File: src/main/java/io/swagslash/gametrackerserver/controller/UserController.java [hunk 9]
```diff
@@ -5,21 +5,36 @@
...
         return userRepository.findById(userPrincipal.getId())
                 .orElseThrow(() -> new ResourceNotFoundException("User", "id", userPrincipal.getId()));
     }
+
+    @PostMapping("/getAgentToken")
+    @PreAuthorize("hasRole('USER')")
+    public String getAgentToken() {
+        return authService.getAgentToken();
+    }
 }
```
Comment (GustavAT) [inline src/main/java/io/swagslash/gametrackerserver/controller/UserController.java]: Please add a new-line

File: src/main/java/io/swagslash/gametrackerserver/dto/webapp/GameDTO.java (hunk 1 above)
Comment (GustavAT) [reply src/main/java/io/swagslash/gametrackerserver/dto/webapp/GameDTO.java:10]: There is no need to add setters if this class is just used to deliver data to the frontend

File: src/main/java/io/swagslash/gametrackerserver/endpoint/webapp/controller/WebappController.java [hunk 10]
```diff
@@ -1,15 +1,41 @@
...
-        return "test";
+    private GameService gameService;
+
+    public WebappController(GameService gameService) {
+        this.gameService = gameService;
+    }
+
+    @GetMapping("/games")
+    @PreAuthorize("hasRole('USER')")
+    public List<GameDTO> findAll() {
```
Comment (GustavAT) [inline src/main/java/io/swagslash/gametrackerserver/endpoint/webapp/controller/WebappController.java]: IMHO, you should not be able to query all games regardless of the user.

File: src/main/java/io/swagslash/gametrackerserver/service/impl/AuthenticationServiceImpl.java [hunk 11]
```diff
@@ -59,6 +70,27 @@ public AuthResponse loginUser(LoginRequest request) {
...
+    public boolean isValidAgentKey(String key) throws BadRequestException {
+        AgentToken token = tokenRepository.findByToken(key);
+        if(token == null) {
+            return false;
+        }
+        return true;
+    }
+
+    @Override
+    public String getAgentToken() throws BadRequestException {
```
Comment (GustavAT) [inline src/main/java/io/swagslash/gametrackerserver/service/impl/AuthenticationServiceImpl.java:83]: Should be `createAgentToken` as you create a new token and do not query it from the `tokenRepository`

**Review by GustavAT — COMMENTED** [review]
Two main concerns:
- url matcher regarding oauth
- `findAll()` games endpoint

Rest LGTM

File: src/main/java/io/swagslash/gametrackerserver/dto/webapp/GameDTO.java (hunk 1 above)
Comment (Monomus) [reply src/main/java/io/swagslash/gametrackerserver/dto/webapp/GameDTO.java:10]: Gamemodes and genres will be added. Didn't exist in the IGDB Wrapper when PR was created.

Comment (Monomus) [reply src/main/java/io/swagslash/gametrackerserver/dto/webapp/GameDTO.java:10]: Getter and Setter are needed for proper Jackson serialzation

File: src/main/java/io/swagslash/gametrackerserver/endpoint/webapp/controller/WebappController.java (hunk 10 above)
Comment (Monomus) [reply src/main/java/io/swagslash/gametrackerserver/endpoint/webapp/controller/WebappController.java]: Method only exists due to testing will be removed

**Review by GustavAT — APPROVED** [review]

**Pull request merged**